src/sentry/integrations/api/endpoints/data_forwarding_details.py

```diff
@@ -0,0 +1,327 @@
+    ) -> bool:
+        if super().has_object_permission(request, view, organization):
+            return True
+
+        if request.method == "PUT":
+            self.determine_access(request, organization)
+            return len(request.access.team_ids_with_membership) > 0
+
+        return False
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Integrations"])
+class DataForwardingDetailsEndpoint(OrganizationEndpoint):
+    owner = ApiOwner.INTEGRATIONS
+    publish_status = {
+        "PUT": ApiPublishStatus.EXPERIMENTAL,
+        "DELETE": ApiPublishStatus.EXPERIMENTAL,
+    }
+    permission_classes = (OrganizationDataForwardingDetailsPermission,)
+
+    def convert_args(
+        self,
+        request: Request,
+        organization_id_or_slug: int | str,
+        data_forwarder_id: int,
+        *args,
+        **kwargs,
+    ):
+        args, kwargs = super().convert_args(request, organization_id_or_slug, *args, **kwargs)
+
+        try:
+            data_forwarder = DataForwarder.objects.get(
+                id=data_forwarder_id,
+                organization=kwargs["organization"],
+            )
+        except DataForwarder.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        kwargs["data_forwarder"] = data_forwarder
+        return args, kwargs
+
+    def _update_data_forwarder_config(
+        self, request: Request, organization: Organization, data_forwarder: DataForwarder
+    ) -> Response:
+        """
+        Request body: {"is_enabled": true, "enroll_new_projects": true, "provider": "segment", "config": {...}, "project_ids": [1, 2, 3]}
+
+        Returns:
+            Response: 200 OK with serialized data forwarder on success,
+                     400 Bad Request with validation errors on failure
+        """
+        data: dict[str, Any] = request.data
+        data["organization_id"] = organization.id
+
+        serializer = DataForwarderSerializer(
+            data_forwarder, data=data, context={"organization": organization}
+        )
+        if serializer.is_valid():
+            data_forwarder = serializer.save()
+            return Response(
+                serialize(data_forwarder, request.user),
+                status=status.HTTP_200_OK,
+            )
+        return self.respond(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def _validate_enrollment_changes(
+        self,
+        request: Request,
+        organization: Organization,
+        data_forwarder: DataForwarder,
+    ) -> tuple[set[int], set[int]]:
+        """
+        Request body: {"project_ids": [1, 2, 3]}
+
+        Validates enrollment changes:
+        - project IDs to be enrolled exist in the organization
+        - User has project:write on projects being enrolled
+        - User has project:write on projects being unenrolled
+
+        Returns:
+            Tuple of (project_ids_to_enroll, project_ids_to_unenroll)
+        """
+        project_ids_new: set[int] = set(request.data.get("project_ids", []))
```

Comment:
nit: I believe you don't need to type annotate all operations, mypy can infer w/o